storage/response_cache.go

```diff
@@ -72,26 +72,32 @@ func (rc *ResponseCache) ClearData(e engine.Engine) error {
 	return err
 }
 
-// GetResponse looks up a response matching the specified cmdID and
-// returns true if found. The response is deserialized into the
-// supplied reply parameter. If no response is found, returns
-// false. If a command is pending already for the cmdID, then this
-// method will block until the the command is completed or the
-// response cache is cleared.
-func (rc *ResponseCache) GetResponse(e engine.Engine, cmdID proto.ClientCmdID) (proto.Response, error) {
+// GetResponse looks up a response matching the specified cmdID. If the
+// response is found, it is returned along with its associated error.
+// If the response is not found, nil is returned for both the response
+// and its error. In all cases, the third return value is a
+// *replicaCorruptionError that is non-nil when the response cache is
+// unable to read from storage.
+func (rc *ResponseCache) GetResponse(e engine.Engine, cmdID proto.ClientCmdID) (proto.Response, error, error) {
```

Comment:
On second thought, dealing with `ResponseWithError` is a bad idea. We lost sight of this somewhere, but the whole point of taking the error off of the Response was to leverage tool support. With `ResponseWithError` we've lost that support again.

I'm going to restore the golint hackery and have this method return two `error`s.